integration_tests/features/circleci_config.feature

```diff
@@ -15,6 +15,167 @@ Feature: Config checking
+      datadog-hello-world:
+        jobs:
+          - datadog-hello-world
+    """
+    When I run `circleci config validate --skip-update-check -c config.yml`
+    Then the exit status should be 0
+    And the output should contain "Config file at config.yml is valid"
+
+  Scenario: Checking a valid config against the k9s server
+    Given a file named "config.yml" with:
+    """
+    version: 2.1
+ 
+    jobs:
+      datadog-hello-world:
+        docker:
+          - image: cimg/base:stable
+        steps:
+          - run: |
+              echo "doing something really cool"
+    workflows:
+      datadog-hello-world:
+        jobs:
+          - datadog-hello-world
+    """
+    When I run `circleci --config-api-host https://k9s.sphereci.com config validate --skip-update-check -c config.yml`
+    Then the exit status should be 0
+    And the output should contain "Config file at config.yml is valid"
+
+  Scenario: Checking a valid config file with an orb
+    Given a file named "config.yml" with:
+    """
+    version: 2.1
+
+    orbs:
+      node: circleci/node@5.0.3
+  
+    jobs:
+      datadog-hello-world:
+        docker:
+          - image: cimg/base:stable
+        steps:
+          - run: |
+              echo "doing something really cool"
+    workflows:
+      datadog-hello-world:
+        jobs:
+          - datadog-hello-world
+    """
+    When I run `circleci config validate --skip-update-check -c config.yml`
+    Then the exit status should be 0
+    And the output should contain "Config file at config.yml is valid"
+
+  Scenario: Checking a valid config file with a private org
+    Given a file named "config.yml" with:
+    """
+    version: 2.1
+
+    orbs:
+      node: circleci/node@5.0.3
+  
+    jobs:
+      datadog-hello-world:
+        docker:
+          - image: cimg/base:stable
+        steps:
+          - run: |
+              echo "doing something really cool"
+    workflows:
+      datadog-hello-world:
+        jobs:
+          - datadog-hello-world
+    """
+    When I run `circleci config validate --skip-update-check --org-id bb604b45-b6b0-4b81-ad80-796f15eddf87 -c config.yml`
+    Then the output should contain "Config file at config.yml is valid"
+    And the exit status should be 0
+
+  Scenario: Checking a valid config file with a non-existant orb
+    Given a file named "config.yml" with:
+    """
+    version: 2.1
+
+    orbs:
+      node: circleci/doesnt-exist@5.0.3
+  
+    jobs:
+      datadog-hello-world:
+        docker:
+          - image: cimg/base:stable
+        steps:
+          - run: |
+              echo "doing something really cool"
+    workflows:
+      datadog-hello-world:
+        jobs:
+          - datadog-hello-world
+    """
+    When I run `circleci config validate --skip-update-check -c config.yml`
+    Then the exit status should be 255
+    And the output should contain "config compilation contains errors"
+
+  Scenario: Checking a valid config file with pipeline-parameters
```

Comment:
nit: would be nice to actually use some of the pipeline-parameters passed in the config string

nit: would be nice to test a case where a pipeline-parameter is used in the config, but not passed in the params.yml

Reply:
Sprinkled in these tests cases and another 1-2 to make sure most cases are covered!